src/common/EIP712/TypedRequestData.ts

```diff
@@ -67,17 +66,19 @@ export default class TypedRequestData implements EIP712TypedData {
       EIP712Domain: EIP712DomainType,
       RelayRequest: RelayRequestType,
       RelayData: RelayDataType,
-      _ForwardRequest: ForwardRequestType
     }
     this.domain = getDomainSeparator(verifier, chainId)
     this.primaryType = 'RelayRequest'
-    this.message = relayRequest
+    //in the signature, all "request" fields are flattened out at the top structure.
+    // other params are inside "relayData" sub-type
+    this.message = {
+      ...relayRequest.request,
+      relayData: relayRequest.relayData
+    }
   }
 }
 
 export const GsnRequestType = {
   typeName: 'RelayRequest',
-  extraParams: 'RelayData relayData',
-  subTypes: 'RelayData(uint256 gasPrice,uint256 pctRelayFee,uint256 baseRelayFee,address relayWorker,address paymaster)',
-  subTypes2: ''
+  typeSuffix: 'RelayData relayData)RelayData(uint256 gasPrice,uint256 pctRelayFee,uint256 baseRelayFee,address relayWorker,address paymaster)',
```

Comment:
Why could it not stay as 'extraParams' and 'subTypes'? You changed the way it signs, did you have to change the API into passing a half-truncated string?

Reply:
if I pass 2 separate items, then I need to make sure they indeed each mean what they meant to mean - but that's exactly the tests I wanted to remove.

Reply:
Can they mean something else with a flattened structure? Honest question.